packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -699,9 +699,10 @@ def send_a2a_transaction(
         :yield: the responses
         """
         stop_condition = self.is_round_ended(self.matching_round.auto_round_id())
-        payload.round_count = cast(
+        round_count = cast(
             SharedState, self.context.state
         ).synchronized_data.round_count
+        object.__setattr__(payload, "round_count", round_count)
```

Comment:
Can we wrap this in a helper method on the base class? Just to expose less internals..

Reply:
I can make it a private method to dismay the developer from using it

Reply:
Not a good idea, then we expose a method which we don't want developers to use. It should only be used in `abstract_round_abci`, which is the case currently, and outside of tests / test tools the `behaviour_utils.py` is the the only place it is used, once.